### PR TITLE
fishPlugin.fish-git-abbr: init at 0.2.1-unstable-2023-06-19

### PR DESCRIPTION
**Pull request by hmajid2301**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/shells/fish/plugins/default.nix [hunk 1]
```diff
@@ -64,3 +66,4 @@ lib.makeScope newScope (self: with self; {
 } // lib.optionalAttrs config.allowAliases {
   autopair-fish = self.autopair; # Added 2023-03-10
 })
+
```
Comment (SuperSandro2000) [inline pkgs/shells/fish/plugins/default.nix:69]:
```suggestion
```

File: pkgs/shells/fish/plugins/git-abbr.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  fetchFromGitHub,
+}:
+buildFishPlugin rec {
+  pname = "fish-git-abbr";
+  version = "dc590a5b9d9d2095f95f7d90608b48e55bea0b0e";
+
+  src = fetchFromGitHub {
+    owner = "lewisacidic";
+    repo = "fish-git-abbr";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/shells/fish/plugins/git-abbr.nix:13]:
```suggestion
  version = "0.2.1-unstable-2023-06-19";

  src = fetchFromGitHub {
    owner = "lewisacidic";
    repo = "fish-git-abbr";
    rev = "dc590a5b9d9d2095f95f7d90608b48e55bea0b0e";
```

**Comment by hmajid2301** [comment]
messed my rebase give me a second :face_exhaling:

**Comment by hmajid2301** [comment]
should be fixed supersandro

File: pkgs/shells/fish/plugins/git-abbr.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}:
+buildFishPlugin rec {
+  pname = "fish-git-abbr";
+  version = "0.2.1-unstable-2023-06-19";
+
+  src = fetchFromGitHub {
+    owner = "lewisacidic";
+    repo = "fish-git-abbr";
+    rev = "dc590a5b9d9d2095f95f7d90608b48e55bea0b0e";
+    sha256 = "sha256-6z3Wr2t8CP85xVEp6UCYaM2KC9PX4MDyx19f/wjHkb0=";
```
Comment (natsukium) [inline pkgs/shells/fish/plugins/git-abbr.nix:14]:
```suggestion
    hash = "sha256-6z3Wr2t8CP85xVEp6UCYaM2KC9PX4MDyx19f/wjHkb0=";
```
nit

File: pkgs/shells/fish/plugins/git-abbr.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  meta = with lib; {
+    description = "Abbreviations for git for the fish shell 🐟.";
```
Comment (natsukium) [inline pkgs/shells/fish/plugins/git-abbr.nix:18]:
```suggestion
    description = "Abbreviations for git for the fish shell 🐟";
```
Could you please remove a trailing period?
https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**